app/components/champ-selection.tsx

```diff
@@ -0,0 +1,59 @@
+import { useEffect, useState } from "react";
+import ChampionsClient from "../_clients/champions-client/champions-client";
+import ChampCard from "./champ-card";
+import {
+  Box,
+  Button,
+  FormControl,
+  FormLabel,
+  HStack,
+  Input,
+} from "@chakra-ui/react";
+
+function ChampSelection({
+  onChampSelection,
+}: {
+  onChampSelection: (championData: ChampionData) => void;
+}) {
+  const [champions, setChampions] = useState<ChampionData[]>(
+    ChampionsClient.getChampionsArray(),
+  );
+
+  const [champSearchValue, setChampionSearchValue] = useState<string>("");
+
+  useEffect(() => {
+    setChampions(ChampionsClient.searchChampions(champSearchValue));
+  }, [champSearchValue]);
```

Comment:
All this could be moved inside the proposed `useChampionClient` hook.

If you move them you can even change the return value for the hook to:
```ts
return {
    champions,
    champSearchValue,
    setChampionSearchValue
};
```

and use it as:
```ts
const { champions, champSearchValue, setChampionSearchValue } = useChampionClient();
```